### ISSUE TITLE
[Question] SC-88, SC-88Pro, etc?

### ISSUE THREAD

**Issue by linoshkmalayil** [nukeykt/Nuked-SC55#89]
I just went through the code.
While I am not that knowledgeable regarding this matter, based on my understanding, it seems possible to extend this to other Sound Canvas models as well.
I am also aware of [emuSC](https://github.com/skjelten/emusc)
I know that he SC-88 rom is not fully decrypted last I checked VOGONS forums. (I am assuming it should be complete by now)
So I am curious is there any limitations and challenges in implementing them?

**Comment by nukeykt** [comment]
PCM and DSP chips are needed to be decapped, traced and analyzed to do it properly, but that will be massive effort

**Comment by linoshkmalayil** [comment]
Reverse engineering the hardware will take a lot of effort.
And decapping obviously means sacrificing one unit.
I don't have the real hardware and beginner level programming skills.
However, I am curious as to how I can contribute.

Also, is there any project focused on correcting sounds in the Sound Canvas VA? (If I am not wrong, it seems like the entire ROM samples are stored in the support DLL SCcore.dll)

**Comment by Kappa971** [comment]
SC-88/88pro emulation would be fantastic but I can't imagine how much work and time it would take and whether it's worth it (games that exceed the 24-28 voices of the SC-55 generally also have CD audio, for example Descent 2).
The mods for Doom I think use midi music composed with the Microsoft GS Wavetable software synthesizer which I think has 32 voices, it would be nice to know if possibly one of these mods exceeds the polyphony of the SC-55 (or mk2).
In any case, SC-88 emulation is welcome if anyone is able to do it (hoping it doesn't require a latest generation Ryzen 9 😅).

**Comment by zaphod77** [comment]
there are definitely midis and game music composed for it, so it's worth the trouble.  

A famous example is Embodiment of Scarlet Devil. ZUN composed a fair amount on the SC88Pro, and often made sc88pro midi versions of later games tunes as well.

The sc88Pro notably has a lot of samples from the jv-1080, which got a lot of desktop composers composing for the thing, especially in Japan.  It's very much worthwhile to emulate.

**Comment by Axis4s** [comment]
i personally wanna see accurate emulated versions of the other units
mostly so i can play with them on reaper and see what they could do.

Ive been getting alot more into how these old hardware synths work.
its very neat that they could be programmed with stuff like sysex for example.
https://github.com/user-attachments/assets/c7a67b86-e553-4ac1-a37b-5dd253480ec7

**Comment by Kappa971** [comment]
> there are definitely midis and game music composed for it, so it's worth the trouble.
> 
> A famous example is Embodiment of Scarlet Devil. ZUN composed a fair amount on the SC88Pro, and often made sc88pro midi versions of later games tunes as well.
> 
> The sc88Pro notably has a lot of samples from the jv-1080, which got a lot of desktop composers composing for the thing, especially in Japan. It's very much worthwhile to emulate.

I don't think that game is very popular outside of Japan, so maybe it's not the best example. On the gaming side, the SC-88 or 88pro would be useful for testing purposes (that's what I think until someone shows me concrete evidence of its use in popular games).
From my ears, I could tell that Shadow Warrior has music composed on the SC-88, but it's recorded on CD.
On the musicians side (which I'm not), such an emulator would certainly be fun.

In any case, we don't know if the work on the SC-88 emulation is stopped or if it's continuing behind the scenes... I'm hoping for the latter anyway.

**Comment by zaphod77** [comment]
dark forces is a western one.

but yes, the sc88pro was very big in japan.  and with sound canvas VA discontinued, it's now quite important from a preservation standpoint to emulate these other synths.

**Comment by zaphod77** [comment]
but sound canvas va won't even be possible to purchase second hand, since it's digital download only, and it may be harder to find vst2 hosts in the future.   I'm well aware that's not really an obstacle to someone sufficiently determined, though.

NO, the filters work fine on VA. it's the previous versions that had borked filters.  Right in the Night GS is just fine in VA, but it breaks on earlier VSC products.

**Comment by linoshkmalayil** [comment]
According the Roland website [here](https://www.rolandcloud.com/news/sound-canvas-va-is-being-discontinued#:~:text=Why%20are%20we%20discontinuing%20Sound,longer%20able%20to%20adequately%20support.)
They are unable to support the framework the SCVA ran on.  (The framework they used seems to run best on Windows 7/8/8.1)
Which seems partially true considering many who complain about activation issues.
I personally had two issues hosting the VST on Elements (UI is broken) and Darkwave Studio (Which down right crash the moment I load the VST)
Currently I am using VSTHost by Hermann Seib.

Since getting accurate low-level emulation SC-88 and SC-88Pro will take time and effort, it's best we give nukeykt and giulioz time to work on it.
Meanwhile, is there any project that decompiles the SC-VA and tries to fix bugs?

**Comment by Kappa971** [comment]
> interesting idea...
> 
> The software seems to run fine on windows 11. The DRM has issues. go figure.
> 
> i know there's a lot of hacking of the yamaha softsynth, but am not aware of any for this one.

You can always search the web for a "special" version of SCVA. It works on Windows 11, for games (I'm not a musician and I've never tried it on music programs).

**Comment by linoshkmalayil** [comment]
I think we need to bring out a separate forum or discussion board to discuss SCVA than to discuss this in Nuked-SC55 git repo.

**Comment by zaphod77** [comment]
fair enough. anyway, more options are always good.

**Comment by mudl0rd** [comment]
> The software seems to run fine on windows 11. The DRM has issues. go figure.

Without saying much, extremely easy to solve (solved it for myself personally). Though not legal, though since its discontinued now, I personally don't see the problem.

**Comment by DMJC** [comment]
My understanding is that Wing Commander 3 might have been done on an SC-88 and Wing Commander IV definitely was. Kilrathi Saga Wing Commander 3 was remastered from the DOS MIDI Files using an SC-88 and the rendered WAV files are included in the game.

**Comment by mudl0rd** [comment]
With those games, how playable are they under DOSBox? Does DOSBox have Windows MIDI API access under DOS?

**Comment by DMJC** [comment]
Wing commander 3 dos has native midi output and supports sound canvas.

**Comment by pachuco** [comment]
> With those games, how playable are they under DOSBox? Does DOSBox have Windows MIDI API access under DOS? I mean...does DOSBox use the MIDI APIs in Windows for GM/GS/XG output?

Under windows? Yes, winmm.

**Comment by Axis4s** [comment]
> Wasn't it that SCVA always had a bug related to filters (TVF, TVA), that they never fixed? I welcome the emulation if it's possible in the future too.

https://dtm.noyu.me/wiki/Roland_Sound_Canvas_VA

SCVA still Shifts the TVF Attack value randomly on a per-note basis.

https://github.com/user-attachments/assets/953bdc66-9be4-4bf7-98f1-09cab771fc94

**Comment by coolkids94** [comment]
> > Wasn't it that SCVA always had a bug related to filters (TVF, TVA), that they never fixed? I welcome the emulation if it's possible in the future too.
> 
> https://dtm.noyu.me/wiki/Roland_Sound_Canvas_VA
> 
> SCVA still Shifts the TVF Attack value randomly on a per-note basis.
> 
>  reaper_0UVLAMmBoz.mp4

is there any way to fix this issue in SCVA or no?

**Comment by MetalMaxMX** [comment]
@giulioz  Wondering but, a reverse engineering of the XP chip would also net us compatibility with the Edirol SD-20, isn't? Since it shares the same chipset and all (see https://chrisoft.org/blog/post/2021-10-10.html)

**Comment by hourai-branch** [comment]
> I don't think that game is very popular outside of Japan, so maybe it's not the best example. 

@Kappa971  [uhhhhhhh..... yeah... sure.](https://www.youtube.com/watch?v=FtutLA63Cp8&list=RDFtutLA63Cp8&start_radio=1&pp=ygUJYmFkIGFwcGxloAcB)
